### PR TITLE
Add e2e sanity test for disk

### PR DESCRIPTION
**Pull request by williazz**
# Description of the issue
Automated sanity test for disk plugin

# Description of changes
Tests that disk metrics are non-negative

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
[Tested on fork here](https://github.com/williazz/amazon-cloudwatch-agent/actions/runs/3563425312/jobs/5991919514)

```
ull_resource.integration_test (remote-exec): 2022/11/28 15:07:57 Running Disk

...

null_resource.integration_test (remote-exec): 2022/11/28 15:10:58 Values are all greater than or equal to zero for disk_free
null_resource.integration_test (remote-exec): 2022/11/28 15:10:58 Values are all greater than or equal to zero for disk_inodes_free
null_resource.integration_test (remote-exec): 2022/11/28 15:10:58 Values are all greater than or equal to zero for disk_inodes_total
null_resource.integration_test (remote-exec): 2022/11/28 15:10:58 Values are all greater than or equal to zero for disk_inodes_used
null_resource.integration_test (remote-exec): 2022/11/28 15:10:58 Values are all greater than or equal to zero for disk_total
null_resource.integration_test (remote-exec): 2022/11/28 15:10:58 Values are all greater than or equal to zero for disk_used
null_resource.integration_test (remote-exec): 2022/11/28 15:10:58 Values are all greater than or equal to zero for disk_used_percent

...

null_resource.integration_test (remote-exec): disk_free           Successful
null_resource.integration_test (remote-exec): disk_inodes_free    Successful
null_resource.integration_test (remote-exec): disk_inodes_total   Successful
null_resource.integration_test (remote-exec): disk_inodes_used    Successful
null_resource.integration_test (remote-exec): disk_total          Successful
null_resource.integration_test (remote-exec): disk_used           Successful
null_resource.integration_test (remote-exec): disk_used_percent   Successful
```

### CODE REVIEW

**Review by khanhntd — DISMISSED** [review]
LGTM.

**Review by SaxyPandaBear — DISMISSED** [review]

File: test/metric/disk.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		"disk_used": {}, 
+		"disk_used_percent": {},
+	}
+}
+
+func (f *DiskMetricValueFetcher) getMetricSpecificDimensions() []types.Dimension {
+	return []types.Dimension{
+		{
+			Name: aws.String("fstype"),
+			Value: aws.String("ext4"),
```
Comment (adam-mateen) [inline test/metric/disk.go:54]: This seems brittle.
What if the host being tested has a different fstype?
Keep in mind all the different OS versions CWA supports, and the various EC2 instance types we could choose.
Will they all have ext4 for the main `/` path?

Comment (SaxyPandaBear) [reply test/metric/disk.go:54]: Good call out. I totally missed that

Comment (williazz) [reply test/metric/disk.go:54]: Thanks for the callout, I'll refactor

Comment (williazz) [reply test/metric/disk.go]: Converting to draft PR to address OS / path / fstype sets

**Pull request closed without merging**